src/main/java/org/threadly/litesockets/NoThreadSocketExecuter.java

```diff
@@ -52,25 +45,20 @@ protected void addThreadExecutorToClient(Client client) {
    * manually.
    */
   public void wakeup() {
-    if(isRunning()) {
-      commonSelector.wakeup();
-    }
-  }
-
-  @Override
-  public void flagNewWrite(Client client) {
-    updateClientOps(client);
+    checkRunning();
+    commonSelector.wakeup();
   }
 
   @Override
-  public void flagNewRead(Client client) {
-    updateClientOps(client);
-  }
-  
-  private void updateClientOps(Client client) {
+  public void setClientOperations(final Client client) {
     ArgumentVerifier.assertNotNull(client, "Client");
-    if(isRunning() && clients.containsKey(client.getChannel())) {
-      if(client.canWrite() && client.canRead()) {
+    if(isRunning() && !clients.containsKey(client.getChannel()) && !client.isClosed() && client.getClientsSocketExecuter() == this) {
```

Comment:
I would vote for #1 as well, since I think that creates a less confusing API (i don't particularly like API's that give you a million ways to do something, because you have to weed through them all to find out what the best way is).

Why not just move the package so you can create protected constructors?  Your going to have a major API change with this anyways.